File: Question_08_07_PermutationsWithoutDuplicatesTests/SolutionTests.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Collections.Generic;
using System.Diagnostics;

namespace Question_08_07_PermutationsWithoutDuplicates.Tests
{
    [TestClass()]
    public class SolutionTests
    {
        [TestMethod()]
        public void GetPermutationsTest()
        {
            string test = "abcd";
            Solution solution = new Solution();
            List<string> result = solution.GetPermutations(test);

            foreach (var perm in result)
            {
                Trace.WriteLine(perm);
            }
        }
    }
}

File: Question_12_01_LastKLinesTests/SolutionTests.cpp
#include "stdafx.h"
#include "CppUnitTest.h"
#include "Solution.h"
#include "TestTrace.h"
#include <string>

using namespace Microsoft::VisualStudio::CppUnitTestFramework;
using namespace Question_12_01_LastKLines;
using namespace NativeTestUtilities;
using namespace std;

//Returns my solution's directory
#define TEST_CASE_DIRECTORY GetDirectoryName(__FILE__)

namespace Question_12_01_LastKLineTests
{
	TEST_CLASS(SolutionTests)
	{
	public:

		TEST_METHOD(PrintLastKLinesTest)
		{
			TestTrace trace;
			Solution solution(trace);

			string filename = string(TEST_CASE_DIRECTORY) + "Readme.txt";

			trace.WriteLine(filename.c_str());
			solution.PrintLastKLines(filename.c_str());
		}

	private:
		//http ://stackoverflow.com/questions/15874723/how-to-open-a-file-from-the-project-in-a-native-c-unit-test-visual-studio
		string GetDirectoryName(string path) {
			const size_t last_slash_idx = path.rfind('\\');
			if (std::string::npos != last_slash_idx)
			{
				return path.substr(0, last_slash_idx + 1);
			}
			return "";
		}

	};
}